src/NServiceBus.AzureStoragePersistence/Subscriptions/Azure/AzureSubscriptionStorage.cs

```diff
@@ -35,34 +38,43 @@ public AzureSubscriptionStorage(string subscriptionTableName, string subscriptio
             };
         }
 
+        static string EncodeTo64(string toEncode)
+        {
+            return Convert.ToBase64String(Encoding.ASCII.GetBytes(toEncode));
+        }
+
+        static string DecodeFrom64(string encodedData)
+        {
+            return Encoding.ASCII.GetString(Convert.FromBase64String(encodedData));
+        }
+
         /// <summary>
-        /// Stores a subscription
+        /// Subscribes the given client to messages of a given type.
         /// </summary>
-        /// <param name="address">The address that is being subscribed to</param>
-        /// <param name="messageTypes">The types of messages that are being subscribed to</param>
-        void ISubscriptionStorage.Subscribe(Address address, IEnumerable<MessageType> messageTypes)
+        /// <param name="subscriber">The subscriber to subscribe</param>
+        /// <param name="messageType">The types of messages that are being subscribed to</param>
+        /// <param name="context">The current context</param>
+        public async Task Subscribe(Subscriber subscriber, MessageType messageType, ContextBag context)
         {
             var table = client.GetTableReference(subscriptionTableName);
 
-
-            foreach (var messageType in messageTypes)
+            try
             {
-                try
+                var subscription = new Subscription
                 {
-                    var subscription = new Subscription
-                    {
-                        RowKey = EncodeTo64(address.ToString()),
-                        PartitionKey = messageType.ToString()
-                    };
-                    var operation = TableOperation.Insert(subscription);
-                    table.Execute(operation);
-                }
-                catch (StorageException ex)
+                    RowKey = EncodeTo64(subscriber.TransportAddress),
+                    PartitionKey = messageType.ToString(),
+                    EndpointName = subscriber.Endpoint.ToString()
+                };
+
+                var operation = TableOperation.Insert(subscription);
+                await table.ExecuteAsync(operation).ConfigureAwait(false);
```

Comment:
No, because we need to wrap it in a `try...catch`. By returning the task we shift that responsibility to the caller.